backend/apps/monkvision/apis/lib/nlp/utils.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * utils.js - NLP Search Utility functions
+ * 
+ * (C) 2022 TekMonks. All rights reserved.
+ */
+
+const numericWords = {
+    units: {
+        'zero': 0, 'one': 1, 'two': 2, 'three': 3, 'four': 4, 'five': 5, 'six': 6, 'seven': 7, 'eight': 8, 'nine': 9, 
+        'ten': 10, 'eleven': 11, 'twelve': 12, 'thirteen': 13, 'fourteen': 14, 'fifteen': 15, 'sixteen': 16, 'seventeen': 17, 
+        'eighteen': 18, 'nineteen': 19, 'twenty': 20, 'thirty': 30, 'forty': 40, 'fifty': 50, 'sixty': 60, 'seventy': 70, 'eighty': 80, 'ninety': 90
+    },
+    magnitudes: {
+        'hundred': 100, 'thousand': 1000, 'million': 1000000, 'billion': 1000000000, 'trillion': 1000000000000
+    },
+    commonWords: ["and"]
+}
+
+const durationUnits = {
+    hours: ["hours", "hrs", "hr", "hour"], minutes: ["minutes", "mins", "min"], seconds: ["seconds", "second"],
+    days: ["days", "day"], weeks: ["weeks", "week"], months: ["months", "month"], years: ["years", "year", "yrs"]
+}
+
+const resourceTerm = ["cluster", "server", "host", "resource"];
+
+/**
+ * Convert Numbers in string to integer
+ * @param {string} sentence - Plain english sentence 
+ * @returns - number (total)
+ */
+function _wordsToNumbers(sentence) {
+    const words = sentence.toString().split(/[\s-]+/);
+    const position = { base: 0, digits: 0 }; 
+    words.forEach((word)=>{ _calculateNumeral(word, position) });
```

Comment:
Never use forEach which uses nested callback functions - only for (... of ...)... or for (... in...)